main/src/ca/uwaterloo/flix/language/phase/jvm/BytecodeInstructions.scala

```diff
@@ -153,17 +153,28 @@ object BytecodeInstructions {
     f
   }
 
-  def IF_ACMPNE(trueBranch: InstructionSet)(falseBranch: InstructionSet): InstructionSet =
-    branch(Opcodes.IF_ACMPNE)(trueBranch)(falseBranch)
+  def IF_ACMPNE(cases: Boolean => InstructionSet): InstructionSet =
```

Comment:
If you replace Boolean by:

```
sealed trait Branch

case object Branch {
  case object TrueBranch
  case object FalseBranch
}
```

then I'll allow it. Otherwise I think its a bit too clever :)